### ISSUE TITLE
Variable for the current page?

### ISSUE THREAD

**Issue by evanmiller** [scmorrison/uzu#46]
I'd like to access the URL slug of the currently rendered page (e.g. `blog/index`, `about`) within a template. Is there a way to do that?

**Comment by samcns** [comment]
Hi @evanmiller 

There isn't a slug built in, but you could add a page variable at the top of the page template:

```yaml
---
slug: 'blog/index'
---
```

And then access it from any partial / theme layout (e.g. `{{ slug }}`).

**Comment by scmorrison** [comment]
Hi @evanmiller 

There isn't a slug built in, but you could add a page variable at the top of the page template:

```yaml
---
slug: 'blog/index'
---
```

And then access it from any partial / theme layout (e.g. `{{ slug }}`).

**Comment by evanmiller** [comment]
Well, I'll explain my problem. I'm trying to add [OpenGraph tags](https://ogp.me) to all of my web pages. So each page needs to contain its canonical URL. In the partial I'd like to do something like:

```
<meta property="og:url" content="https://coolwebsite.com/{{ page }}.html">
```

Adding a slug variable to each template would be relatively cumbersome, and also easy to get out of sync if the template moves.

It's not an urgent need but I just wanted to see if there was an easy solution here.

**Comment by scmorrison** [comment]
Hi @evanmiller 

I'll take a look and see what I can come up with. I think it is a good idea and would be pretty useful.

Thanks,
Sam

**Comment by thowe** [comment]
Was this ever implemented?

**Comment by scmorrison** [comment]
@thowe Yes, there is a page hash now, https://github.com/scmorrison/uzu#page-variables. `page.id` will return the unique path, minus extension, for the page (e.g. `/index`, `/path/to/page`).
